test/integration/loader.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import re
+import sys
+
+re_empty = re.compile("^$")
+re_req = re.compile("^>>> (.+)$")
+re_rsp = re.compile("^<<< (.+)$")
+re_stat = re.compile("^\+\+\+ (.+)$")
+
+
+def split_metrics(line):
+    """split a line of metric deltas into a dict, for example:
+
+    'get +1, get_hit +1' -> {'get':1, 'get_hit':1}
+    'request_free -1, request_parse +1' -> {'request_free':-1, 'request_parse':1}
+    """
+    metrics = line.split(',')
+    d = {}
+    for m in metrics:
+        name,delta = m.strip().split(' ')
+        d[name.strip()] = int(delta)
+
+    return d
+
+
+def load_seq(fname):
+    """Load the test (a sequence of commands and asserts) from a file.
+    Each command contains one or more lines of request, leading with '>>> ', and
+    one or more lines of response, leading with '<<< '. Commands are separated
+    by an empty line.
+    """
+    lines = open(fname).readlines()
+    if not re_empty.match(lines[-1]):  # ensure an empty line at the end
+      lines.append('\n')
+
+    seq = []
```

Comment:
this structure is a bit more complicated than I expected... why don't you just make an array seq with three types of values, either request, response or stats and read it line by line? No need to aggregate them in a single sequence of all three... this is also prone to errors if you declare a req, then a stat, then a req, since both req will be executed before the stat (unless there's a blank line in between!).